src/App.tsx

```diff
@@ -1,10 +1,67 @@
-import React from 'react';
+import React, {useState} from 'react';
+import axios from 'axios';
 import './App.css';
+import UserProfile from './UserProfile';
 
+
+const BASE_URL:string = "https://api.github.com/users/"; 
+let timeout:any;
 function App() {
+  const [userName, setUserName] = useState<string>("");
+  const [userData, setUserData] = useState<object>();
+  const [error, setError] = useState<string>("");
+ 
+  const getData = async () => {
+    const CancelToken = axios.CancelToken;
+      const source = CancelToken.source();
+    
+      await axios.get(BASE_URL + userName, {cancelToken: source.token})
+      .then((response) => {
+        if(userName === response.data.login.toLowerCase()){
+          const data = response.data;
+          setUserData(data);
+          setError("");
+        }
+       })
+      .catch((error) => {
+        setUserData({});
+        if(error.response){
+          setError(error.response.statusText);
+        } else if (error.request){
+          setError(error.request.XMLHttpRequest.statusText)
```

Comment:
Error from request??

Reply:
There is limit for 60 request per hour in this API. So to counter the part that if a user makes more than 60 requests, he will get a error will requesting and it will be displayed.